packages/db-model/src/db.ts

```diff
@@ -78,97 +77,65 @@ async function withTimestamp(connection: Knex, table: string, params: any = {}):
   const exists = await connection.schema.hasColumn(table, 'updated_at');
 
   if (!exists || (exists && params.updatedAt)) {
-    return object.toSnakeCase(params);
+    return params;
   }
 
-  return { ...object.toSnakeCase(params), updated_at: connection.fn.now() };
+  return { ...params, updated_at: connection.fn.now() };
 }
 
 /**
  * Finds a record based on the params.
- * Returns null if no results were found.
  *
  * @param {Knex} connection
  * @param {string} table
  * @param {object} [params={}]
+ * @param {Function} callback
  * @param {Knex.Transaction} [trx]
- * @returns {(Promise<T | null>)}
+ * @returns {Knex.QueryBuilder}
  */
-export async function get<T>(
+export function find<T>(
   connection: Knex,
   table: string,
   params: object = {},
+  callback?: any,
```

Comment:
Why do we need a callback here? It's a simple function.

Reply:
Callbacks are helpful when we're required to add a custom select or append where clauses or can be used for any modification to the default query builder generated by the db-model.

`const evaluation = await Evaluation.findByIdOrFail(id, appendAuthFilter);`
```
function appendAuthFilter(qb) {
  qb.where(user, userObj);
}
```


`const evaluation = await Evaluation.findByIdOrFail(id, customSelect);`
```
function customSelect(qb) {
  qb.clearSelect();
  qb.select('a, b, c');
}
```

cc @mesaugat

Reply:
If they need custom query building then using quick functions like find or findbyIdOrFail won't make sense too - it's better that they do with the query builder directly so that they can write the whole logic at once - it's flexible, consistent and clean too.

Something like this:
```ts
const result = User.buildQuery<UserWithSettings>(
  qb => qb
    .select('a, b, c')
    .from(this.table)
    .innerJoin('employees AS emp', 'em.manager_employee_id', 'emp.id')
    .leftOuterJoin('user_accounts AS ua', 'em.manager_employee_id', 'ua.employee_id')
    .whereIn('em.manager_employee_id', ids)
)
```

It's better if we can leave the functions like find, findOrFail and others simple for trivial and oneliner use cases only because they can't fit in the complex cases either.